google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta1/stub/EnhancedBigQueryStorageStub.java

```diff
@@ -45,51 +47,55 @@
-        .setRetrySettings(settings.splitReadStreamSettings().getRetrySettings());
-
-    BigQueryStorageStubSettings baseSettings = baseSettingsBuilder.build();
-    ClientContext clientContext = ClientContext.create(baseSettings);
-    GrpcBigQueryStorageStub stub = new GrpcBigQueryStorageStub(baseSettings, clientContext);
-    return new EnhancedBigQueryStorageStub(stub);
+    try {
+      BigQueryStorageStubSettings.Builder baseSettingsBuilder =
+          BigQueryStorageStubSettings.newBuilder()
+              .setTransportChannelProvider(settings.getTransportChannelProvider())
+              .setEndpoint(settings.getEndpoint())
+              .setHeaderProvider(settings.getHeaderProvider())
+              .setCredentialsProvider(settings.getCredentialsProvider())
+              .setStreamWatchdogCheckInterval(settings.getStreamWatchdogCheckInterval())
+              .setStreamWatchdogProvider(settings.getStreamWatchdogProvider());
+
+      // CreateReadSession is a simple pass-through.
+      baseSettingsBuilder
+          .createReadSessionSettings()
+          .setRetryableCodes(settings.createReadSessionSettings().getRetryableCodes())
+          .setRetrySettings(settings.createReadSessionSettings().getRetrySettings());
+
+      // ReadRows is a simple pass-through.
+      baseSettingsBuilder
+          .readRowsSettings()
+          .setRetryableCodes(settings.readRowsSettings().getRetryableCodes())
+          .setRetrySettings(settings.readRowsSettings().getRetrySettings())
+          .setResumptionStrategy(settings.readRowsSettings().getResumptionStrategy())
+          .setIdleTimeout(settings.readRowsSettings().getIdleTimeout());
+
+      // BatchCreateReadSessionStreams is a simple pass-through.
+      baseSettingsBuilder
+          .batchCreateReadSessionStreamsSettings()
+          .setRetryableCodes(settings.batchCreateReadSessionStreamsSettings().getRetryableCodes())
+          .setRetrySettings(settings.batchCreateReadSessionStreamsSettings().getRetrySettings());
+
+      // FinalizeStream is a simple pass-through.
+      baseSettingsBuilder
+          .finalizeStreamSettings()
+          .setRetryableCodes(settings.finalizeStreamSettings().getRetryableCodes())
+          .setRetrySettings(settings.finalizeStreamSettings().getRetrySettings());
+
+      // SplitReadStream is a simple pass-through.
+      baseSettingsBuilder
+          .splitReadStreamSettings()
+          .setRetryableCodes(settings.splitReadStreamSettings().getRetryableCodes())
+          .setRetrySettings(settings.splitReadStreamSettings().getRetrySettings());
+
+      BigQueryStorageStubSettings baseSettings = baseSettingsBuilder.build();
+      ClientContext clientContext = ClientContext.create(baseSettings);
+      GrpcBigQueryStorageStub stub = new GrpcBigQueryStorageStub(baseSettings, clientContext);
+      return new EnhancedBigQueryStorageStub(stub);
+    } catch (Exception e) {
+      throw newBigQueryStorageException(e);
```

Comment:
I'm not quite sure I understand the plumbing here -- this looks like it would wrap exceptions which occur when the gRPC connection is initially established, but not errors which occur when subsequently making calls. Am I missing something?